src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -1635,6 +1660,14 @@ private void getCallInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_RESO
 
             pResult->methodFlags = FilterNamedIntrinsicMethodAttribs(pResult->methodFlags, methodToCall);
 
+            var targetDetails = _compilation.TypeSystemContext.Target;
+            if (targetDetails.Architecture == TargetArchitecture.X86
+                && targetDetails.OperatingSystem == TargetOS.Windows
+                && targetMethod.IsNativeCallable)
+            {
+                throw new NotImplementedException("ReadyToRun: References to methods with NativeCallableAttribute not implemented");
```

Comment:
AFAIK this is going to tear down the compilation process. Throwing `RequiresRuntimeJitException` would be more appropriate because that one gets handled.

(Same comment for the NotImplement in CorInfoImpl.cs)